src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceDeserializer.cs

```diff
@@ -623,7 +692,8 @@ private object ReadNestedResourceInline(ODataResourceWrapper resourceWrapper, IE
                         Error.Format(SRResources.MappingDoesNotContainResourceType, structuredType.FullName()));
                 }
 
-                nestedReadContext.ResourceType = typeof(List<>).MakeGenericType(clrType);
+                nestedReadContext.ResourceType = (readContext.IsDeltaOfT && resourceSetWrapper.ResourceSetType == ResourceSetType.DeltaResourceSet)
+                ? typeof(Delta<>).MakeGenericType(clrType) : typeof(List<>).MakeGenericType(clrType);
```

Comment:
>Delta [](start = 25, length = 5)

why is it Delta<T>, not DeltaSet<T> ?
do you have the test cases covering the nested deltaset?

Reply:
Yea its DeltaSet, It used to work because in resourcesetdeserializer it creates type of DeltaSEt<> in case of untyped even if resourcetype set here is delta 
Yes  I have  testcases covering that